### DISCUSSION TITLE
Installing MFRONT for Abaqus

### DISCUSSION THREAD

**Discussion by HartreeFock** [thelfer/tfel#433] [General]
Hi to everyone,

My name is Daniele I am a Process Engineer at the Advanced Forming Research Centre in UK. I was aware of MFRONT since my post doc at the University of Glasgow where we worked to implement it in MOFEM and was really nice to test it.

At the moment I am working with commercial codes, Abaqus 2022 using both Explicit and Standard solver. My machine work on Windows and I have all the compilers for both C++ and Fortran installed since I usually script in Fortran.

I wanted to start to work with MFRONT but I am finding difficult to understand which is the correct procedure to install it and use it in my environment. I tried to look for the documentation, and I am not went really far. Is there anything that I missed or there is anyone able to give me a good direction where to start?

Thank you very much in advance.

Best

Daniele

**Comment by thelfer** [comment]
Dear Daniele,

Thanks for your interest.

I know that some people did successfuly used MFront and Abaqus on Windows. The trouble with windows is mixing environments. Furthermore, TFEL-4.0.x is no more compatible with Visual Studio since this compiler does not support some valid C++ syntaxes.

If you are adventurous, you may want to install TFEL with the intel compilers using `cmake`: https://thelfer.github.io/tfel/web/install.html#using-the-cmake-build-system

In your case, you have to do something like this:

~~~~
cmake <path to TFEL sources>  <compiler options> -Denable-abaqus=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=<where_to_install_TFEL>
~~~~

where <compiler options> specify that you want to use the intel compilers: see https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-0/use-cmake-with-the-compiler.html for `icx`. There must be an alternative if you plan to use the legacy intel compilers.

**Reply by HartreeFock** [reply]
Dear Thomas,

many thanks for your reply. I am always keen to learn :) and in this specific case I must do it since the benefits of having MFRONT outclass the trouble in installing it. I will update here my progress.

**Reply by thelfer** [reply]
Please do.
